Graphics/Win32/Window.hsc

```diff
@@ -261,12 +281,29 @@ foreign import WINDOWS_CCONV "windows.h CreateWindowExW"
 
 ----------------------------------------------------------------
 
+-- | Delegates to the standard default window procedure, but if it receives the
+-- 'wM_NCDESTROY' message it first frees the window closure to allow the
+-- closure and any objects it closes over to be garbage collected. 'wM_NCDESTROY' is
+-- the last message a window receives prior to being deleted.
 defWindowProc :: Maybe HWND -> WindowMessage -> WPARAM -> LPARAM -> IO LRESULT
-defWindowProc mb_wnd msg w l =
+defWindowProc mb_wnd msg w l = do
+  when (msg == wM_NCDESTROY) (maybe (return ()) freeWindowProc mb_wnd)
```

Comment:
The rest of the changes I'm fine with, this one I'm a bit concerned about. `Win32` is supposed to be a very low level binding. I'm concerned about it doing anything automatically for you that API doesn't do and that you can't opt out of.

so how about instead having this behavior under a function with a different name like `defWindowProcSafe` or similar. And a comment here telling people about it.

Often times people look at the MSDN documentation for what the function does and not the haddock.

Reply:
I appreciate the intent of bindings is to be low level. The reason I decided to propose this change was to cancel out the other excursion from low level bindings that come from the variations in `registerWindowClass` (which doesn't take a window closure/window procedure, unlike the C equivalent) and `setWindowClosure` (which doesn't really correspond to anything).

However, I'm happy to accept your proposal. I will make a consequential change to the example as well.

Reply:
Yeah I think the change to return the handle is fine it's just this one since you can't override it. Thanks!